code_files/file_parser.py

```diff
@@ -0,0 +1,55 @@
+import os
+import csv
+import zipfile
+from weather_readings import WeatherReading
+
+class Parser:
+    def __init__(self, extract_to):
+        self.extract_to = extract_to
+        self.weather_readings = []
+
+    def extract_zip(self):
+        """Extracts files from a weatherfiles archive.
+
+        Args:
+            extract_to (str): The path to the directory where the extracted files will be placed.
+
+        Raises:
+            RuntimeError: If there are any errors during extraction.
+        """
+        try:
+            with zipfile.ZipFile("weatherfiles.zip", "r") as zip_ref:
+                zip_ref.extractall(self.extract_to)
+        except zipfile.ZipFile as e:
+            raise RuntimeError(f"Error extracting files: {e}")
+    
+    def parse_weather_file(self, path):
+        """Parses a TXT file containing weather data.
+
+        This method reads a TXT file at the specified path and populates the
+        weather_readings list of the class with WeatherReading objects. Each
+        object represents a day's weather data.
+        """
+        with open(path, 'r') as file:
+            reader = csv.DictReader(file)
+            for row in reader:
+                try:
+                    reading = WeatherReading(
+                        date=row[list(row.keys())[0]],
+                        max_temp=int(row['Max TemperatureC']),
+                        min_temp=int(row['Min TemperatureC']),
+                        max_humidity=int(row['Max Humidity']),
+                        mean_humidity=int(row[' Mean Humidity'])
+                    )
+                    self.weather_readings.append(reading)
+                except ValueError:
+                    continue
```

Comment:
Let's move these calculations to `WeatherReading.validate_reading` and avoid using try except through out whole of this task.

Reply:
sure